internal/connector/connector.go

```diff
@@ -230,20 +225,25 @@ func (cw *wrapper) runServer(ctx context.Context, serverCred *tlsV1.Credential)
 
 // C returns a ConnectorClient that the caller can use to interact with a locally running connector.
 func (cw *wrapper) C(ctx context.Context) (types.ConnectorClient, error) {
+	// Check to see if we have a client already
+	cw.mtx.RLock()
```

Comment:
`A` - First call to `C`
`B` - Second call to `C`
`C` - Third call to `C`

`A` happens, gets `RLock`
`B` happens, gets `RLock`

`A` & `B` see no client.

`A` gets `Lock` first.
`B` blocked on acquiring `Lock` by `A`'s lock on `Lock`

`C` happens, blocked on acquiring `RLock` by `A`'s lock on `Lock`

`A` runs through and creates client.
`B` unblocked, acquires `Lock` sees client created and early returns.
`C` unblocked from acquiring `RLock`, sees client and early returns.

How is this different than if you just got a `Lock` on line 229 instead of an `RLock`?

`A` acquires `Lock`
`B` blocked on acquiring `Lock`
`C` blocked on acquiring `Lock`

`A`sees no client and continues on and creates a client.
`B` unblocked,  acquires `Lock`, early returns on seeing client
`C` unblocked, acquires `Lock`, early returns on seeing client.

No `Rlock` needed?

Just trying to understand, thank you!

Reply:
Oh I think understand, it's so that multiple calls can get a client once it's been created without one locking the other, but getting a client is blocked while one is being created.